packages/kbn-search-connectors/lib/update_connector_service_type.ts

```diff
@@ -6,42 +6,29 @@
  * Side Public License, v 1.
  */
 
+import { Result } from '@elastic/elasticsearch/lib/api/types';
 import { ElasticsearchClient } from '@kbn/core/server';
-import { i18n } from '@kbn/i18n';
-
-import { CONNECTORS_INDEX, fetchConnectorById } from '..';
-
-import { ConnectorDocument, ConnectorStatus } from '../types/connectors';
 
 export const updateConnectorServiceType = async (
   client: ElasticsearchClient,
   connectorId: string,
   serviceType: string
 ) => {
-  const connectorResult = await fetchConnectorById(client, connectorId);
-
-  if (connectorResult?.value) {
```

Comment:
Is it no longer necessary to confirm the connector doc exists first?

Reply:
Correct, if document doesn't exist we return 404 from Connector API. 

In status quo the check makes sure that we don't write partial update to the index directly.